src/components/FAB.js

```diff
@@ -0,0 +1,60 @@
+import React, {memo} from 'react';
+import {Pressable, Animated} from 'react-native';
+import PropTypes from 'prop-types';
+import {PlusIcon} from './Expensicons';
+import styles from '../styles/styles';
+import colors from '../styles/colors';
+import themeColors from '../styles/themes/default';
+
+const AnimatedPlusIcon = Animated.createAnimatedComponent(PlusIcon);
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
+
+const propTypes = {
+    // Callback to fire on request to toggle the FAB
+    onPress: PropTypes.func.isRequired,
+
+    // Animated value to control the animation
+    animatedValue: PropTypes.instanceOf(Animated.Value).isRequired,
+
+    // Temporary prop to be able to test the component before implementing it
+    isHidden: PropTypes.bool,
+};
+
+const defaultProps = {
+    isHidden: true,
+};
+
+const FAB = ({onPress, animatedValue, isHidden}) => {
+    const rotate = animatedValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: ['0deg', '135deg'],
+    });
+    const backgroundColor = animatedValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: [colors.green, colors.transparent],
+    });
+    const fill = animatedValue.interpolate({
+        inputRange: [0, 1],
+        outputRange: [colors.white, themeColors.icon],
+    });
+
+    if (isHidden) {
+        return null;
+    }
+
+    return (
+        <AnimatedPressable
+            onPress={onPress}
+            style={[
+                styles.floatingActionButton,
+                {transform: [{rotate}], backgroundColor},
+            ]}
+        >
+            <AnimatedPlusIcon fill={fill} />
+        </AnimatedPressable>
+    );
+};
+
+FAB.defaultProps = defaultProps;
+FAB.propTypes = propTypes;
+export default memo(FAB);
```

Comment:
Just curious here. Why is it better to use `memo()` in this case?

Reply:
You got me here. I started creating the FAB component copying the Avatar component (at random), the memo just came with it.

With such a small component, I don't believe there would be any difference in performance. I removed it for the next commit.